reader_test.go

```diff
@@ -0,0 +1,47 @@
+package idletiming
+
+import (
+	"io"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReader(t *testing.T) {
+	l, err := net.Listen("tcp", ":0")
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer l.Close()
+
+	go func() {
+		for {
+			conn, err := l.Accept()
+			if err != nil {
+				return
+			}
+			go func() {
+				for {
+					conn.Write([]byte("aaaaaaaaaaaaaa"))
```

Comment:
How about adding a small delay (but higher than reader's timeout) before each `Write`, to demo that the reader can actually tolerate timeouts? Below patch works to me.

```diff
diff --git a/reader_test.go b/reader_test.go
index 42bd873..53b06ed 100644
--- a/reader_test.go
+++ b/reader_test.go
@@ -24,6 +24,7 @@ func TestReader(t *testing.T) {
                        }
                        go func() {
                                for {
+                                       time.Sleep(3 * time.Millisecond)
                                        conn.Write([]byte("aaaaaaaaaaaaaa"))
                                }
                        }()
@@ -36,8 +37,8 @@ func TestReader(t *testing.T) {
        }
        defer conn.Close()

-       r := NewReader(conn, 250*time.Millisecond)
-       b := make([]byte, 100000)
+       r := NewReader(conn, 2*time.Millisecond)
+       b := make([]byte, 1000)
        n, err := io.ReadFull(r, b)
        if !assert.NoError(t, err) {
                return
```

Reply:
Great idea, done!